src/oracleFactory/OracleFactory.ts

```diff
@@ -53,7 +58,11 @@ class IExecOracleFactory {
    * @param options Optional configuration options OracleFactory.
    */
   constructor(
-    ethProvider: Eip1193Provider | Web3SignerProvider,
+    ethProvider:
+      | Eip1193Provider
+      | Web3SignerProvider
+      | AbstractSigner
+      | BrowserProvider,
```

Comment:
In dataprotector-sdk and web3mail-sdk we have:
```
    ethProvider?:
      | AbstractProvider
      | AbstractSigner
      | Eip1193Provider
      | Web3SignerProvider
      | string,
```

Should we add AbstractProvider here in oracle-factory?

Should we also add BrowserProvider in dataprotector-sdk and web3mail-sdk?